cpp/src/transform/cudf_to_arrow.cpp

```diff
@@ -0,0 +1,180 @@
+
+    column_view input_view = (tmp_column != nullptr) ? tmp_column->view() : input;
+
+    auto mask_buffer = fetch_mask_buffer(input_view, ar_mr);
+    std::vector<std::shared_ptr<arrow::Array>> child_arrays;
+
+    for (size_type i = 0; i < input_view.num_children(); i++) {
+      auto c = input_view.child(i);
+      child_arrays.emplace_back(
+        type_dispatcher(c.type(), dispatch_to_arrow{}, c, c.type().id(), ar_mr));
+    }
+
+    if (std::is_same<T, cudf::string_view>::value) {
+      if (child_arrays.size() == 0) {
+        return std::make_shared<arrow::StringArray>(0, nullptr, nullptr);
+      }
+      auto offset_buffer = child_arrays[0]->data()->buffers[1];
+      auto data_buffer   = child_arrays[1]->data()->buffers[1];
+      return std::make_shared<arrow::StringArray>(static_cast<int64_t>(input_view.size()),
+                                                  offset_buffer,
+                                                  data_buffer,
+                                                  mask_buffer,
+                                                  static_cast<int64_t>(input_view.null_count()));
+    } else if (std::is_same<T, cudf::dictionary32>::value) {
+      // Update indices with bit mask buffer
+      auto indices    = to_arrow_array(type_id::INT32,
+                                    static_cast<int64_t>(input_view.size()),
+                                    child_arrays[0]->data()->buffers[1],
+                                    mask_buffer,
+                                    static_cast<int64_t>(input_view.null_count()));
+      auto dictionary = child_arrays[1];
+      return std::make_shared<arrow::DictionaryArray>(
+        arrow::dictionary(indices->type(), dictionary->type()), indices, dictionary);
+    } else {
+      auto offset_buffer = child_arrays[0]->data()->buffers[1];
+      auto data          = child_arrays[1];
+      return std::make_shared<arrow::ListArray>(arrow::list(data->type()),
+                                                static_cast<int64_t>(input_view.size()),
+                                                offset_buffer,
+                                                data,
+                                                mask_buffer,
+                                                static_cast<int64_t>(input_view.null_count()));
+    }
+  }
+
+  template <typename T>
+  std::enable_if_t<(!is_fixed_width<T>()) and (!is_compound<T>()), std::shared_ptr<arrow::Array>>
+  operator()(column_view input_view, cudf::type_id id, arrow::MemoryPool* ar_mr)
+  {
+    CUDF_FAIL("Only fixed width and compund types are supported");
+  }
+};
+
+}  // namespace detail
+
+std::shared_ptr<arrow::Table> cudf_to_arrow(table_view input,
+                                            std::vector<std::string> const& column_names,
+                                            arrow::MemoryPool* ar_mr)
+{
+  auto num_columns = input.num_columns();
+
+  CUDF_EXPECTS((column_names.size() == num_columns),
```

Comment:
The temporary is never used other than here. Just use it directly. 
```suggestion
  CUDF_EXPECTS((column_names.size() == input.num_columns()),
```